pkg/quota/evaluator/core/persistent_volume_claims.go

```diff
@@ -105,45 +81,13 @@ type pvcEvaluator struct {
 
 // Constraints verifies that all required resources are present on the item.
 func (p *pvcEvaluator) Constraints(required []api.ResourceName, item runtime.Object) error {
-	pvc, ok := item.(*api.PersistentVolumeClaim)
-	if !ok {
-		return fmt.Errorf("unexpected input object %v", item)
-	}
-
-	// these are the items that we will be handling based on the objects actual storage-class
-	pvcRequiredSet := append([]api.ResourceName{}, pvcResources...)
-	if storageClassRef := helper.GetPersistentVolumeClaimClass(pvc); len(storageClassRef) > 0 {
-		pvcRequiredSet = append(pvcRequiredSet, ResourceByStorageClass(storageClassRef, api.ResourcePersistentVolumeClaims))
-		pvcRequiredSet = append(pvcRequiredSet, ResourceByStorageClass(storageClassRef, api.ResourceRequestsStorage))
-	}
-
-	// in effect, this will remove things from the required set that are not tied to this pvcs storage class
-	// for example, if a quota has bronze and gold storage class items defined, we should not error a bronze pvc for not being gold.
-	// but we should error a bronze pvc if it doesn't make a storage request size...
-	requiredResources := quota.Intersection(required, pvcRequiredSet)
-	requiredSet := quota.ToSet(requiredResources)
-
-	// usage for this pvc will only include global pvc items + this storage class specific items
-	pvcUsage, err := p.Usage(item)
-	if err != nil {
-		return err
-	}
-
-	// determine what required resources were not tracked by usage.
-	missingSet := sets.NewString()
-	pvcSet := quota.ToSet(quota.ResourceNames(pvcUsage))
-	if diff := requiredSet.Difference(pvcSet); len(diff) > 0 {
-		missingSet.Insert(diff.List()...)
-	}
-	if len(missingSet) == 0 {
-		return nil
-	}
-	return fmt.Errorf("must specify %s", strings.Join(missingSet.List(), ","))
+	// no-op for persistent volume claims
+	return nil
```

Comment:
This change isn't obvious to me.

Reply:
the existing behavior was a bug imho.  it was basically copying a pattern from pods that did not make sense for pvcs that basically verified each pvc made a storage request.